include/secp256k1_schnorrsig.h

```diff
@@ -0,0 +1,118 @@
+#ifndef SECP256K1_SCHNORRSIG_H
+#define SECP256K1_SCHNORRSIG_H
+
+/** This module implements a variant of Schnorr signatures compliant with
+ * BIP-schnorr
+ * (https://github.com/sipa/bips/blob/bip-schnorr/bip-schnorr.mediawiki).
+ */
+
+/** Opaque data structure that holds a parsed Schnorr signature.
```

Comment:
As the Schnorr signature BIP defines the actual byte representation of signatures, there is no distinction between "invalid because incorrect values" and "invalid because incorrect serialization" (as is the case with ECDSA and DER encoding).

Because of that, there seems to not be any strict need for separately exposed parse/serialize functions. That is unless we expect operations besides sign/verify that operate on signatures, which one is expected to use without intervening serialization step. In that case we may opt to later change the representation inside to be a `secp256k1_fe_storage` and a `secp256k1_scalar`.

Reply:
We cannot use `secp256k1_fe_storage` or `secp256k1_scalar` because the size of these objects is not exported, so it would prevent users building them on the stack.

Eliminating parse/serialize operations will prevent any future changes we might want to do that could use a more efficient format. On the other hand, I can't imagine what future changes those might be.

Reply:
Look at how `secp256k1_pubkey_load` works :)

And yes, I can't imagine what those operations would be either. I'd like to make sure we're not committing to an unnecessarily complex API without reason.

Reply:
Makes sense to me to remove the parse/serialize functions. In the event that there's a better representation it could be its own struct.

Reply:
> That is unless we expect operations besides sign/verify that operate on signatures, which one is expected to use without intervening serialization step.

There are definitely operations besides sign/verify that operate on signatures, for example verifying a sign-to-contract commitment as implemented here https://github.com/bitcoin-core/secp256k1/pull/572/files#diff-b19c5ee427283d4d82bc5beb4e2f4777R202

Reply:
We shouldn't speculate on future needs that we're not sure of too much-- if something useful turns up we can change the API.  A guessed API might not be right in any case.